### ISSUE TITLE
Ordering of MMR in chained representation models has no effect

### ISSUE THREAD

**Issue by Quadrophonia** [MaartenGr/BERTopic#2394]
### Have you searched existing issues?  🔎

- [x] I have searched and found no existing issues

### Desribe the bug

Maarten, thank you so much for your work on BERTopic.

[Similar to this issue](https://github.com/MaartenGr/BERTopic/issues/1224), I have a question regarding the ordering of representation models in a [chained representation model setup](https://maartengr.github.io/BERTopic/getting_started/representation/representation.html#chain-models). Here I am trying to get the redundancy benefit of the MMR representation, with the keyword power of KeyBERT. However, I am finding that I lose the MMR benefits when I put it in any kind of chain.

MMR alone yields the following topics, for example, with low redundancy but also small stopwords:

```
mmr = MaximalMarginalRelevance(diversity=0.3)
representation_model = [mmr]
```
> Topic 1: {car; bike; my; in; engine}
Topic 4: {encryption; clipper; chip; keys; government}

KeyBERTInspired alone yields the following topics, with higher redundancy and no small stopwords:

```
kbmr = KeyBERTInspired()
representation_model = [kbmr]
```

> Topic 1: {riding; motorcycle; bikes; bike; ride}
Topic 4: {encryption; cryptography; crypto; encrypted; security}

However, if I chain MMR *after*, say, KeyBERTInspired or PartOfSpeech, I lose the redundancy benefits and the results look identical to those obtained using *only the KeyBERTInspired model*:

```
mmr = MaximalMarginalRelevance(diversity=0.3)
kbmr = KeyBERTInspired()
representation_model = [kbmr,mmr]
```

> Topic 1: {riding, motorcycle, bikes, bike, ride}
Topic 4: {encryption; cryptography; crypto; encrypted; security}

If I chain MMR *before* KeyBERTInspired, the results are identical to the above. 

```
mmr = MaximalMarginalRelevance(diversity=0.3)
kbmr = KeyBERTInspired()
representation_model = [mmr,kbmr]
```

> Topic 1: {riding, motorcycle, bikes, bike, ride}
Topic 4: {encryption; cryptography; crypto; encrypted; security}

To be honest, I doubt this is a bug: more likely, I'm doing something wrong, but I don't know what. I've tried this with both v16.4 and 17.0. Any help would be appreciated! Thank you again!

### Reproduction

```python
from bertopic import BERTopic
from sklearn.datasets import fetch_20newsgroups
from umap import UMAP
from bertopic.representation import MaximalMarginalRelevance, PartOfSpeech, KeyBERTInspired

docs = fetch_20newsgroups(subset='all', remove=('headers', 'footers', 'quotes'))["data"]

seed_topic_list = [["drug", "cancer", "drugs", "doctor"],
["windows", "drive", "dos", "file"],
["space", "launch", "orbit", "lunar"]]

umap_model = UMAP(n_neighbors=20, n_components=5, min_dist=0.1, metric='cosine', random_state=42)

# Specify the chained represenation model
mmr = MaximalMarginalRelevance(diversity=0.3) 
posrm = PartOfSpeech("en_core_web_sm")
kbmr = KeyBERTInspired()

representation_model = [mmr, kbmr] 

topic_model = BERTopic(
    umap_model=umap_model,
    min_topic_size=30, 
    embedding_model="all-MiniLM-L6-v2",
    representation_model = representation_model,
).fit(docs)

topics, probs = topic_model.fit_transform(docs)

fig = topic_model.visualize_barchart()
fig.show() 

print(topic_model.get_topic_info())

```

### BERTopic Version

0.17

**Comment by AnasNeji** [comment]
Hello @Quadrophonia

If you want to have topics without taking into account stop words you can take care of that in the CountVectorizer level

you can do something like this:

vectorizer_model = CountVectorizer(stop_words="english") ## you can also pass a customized stop words list

topic_model = BERTopic(
    umap_model=umap_model,
    min_topic_size=30, 
    embedding_model="all-MiniLM-L6-v2",
    representation_model = representation_model,
    vectorizer_model=vectorizer_model
).fit(docs)

**Comment by Quadrophonia** [comment]
Hi @AnasNeji and thanks for reading my post! Yes, my full code uses the CountVectorizer to manage stop words. However, the underlying ordering problem in my original post still affects my analysis. Thanks for your comment!

**Comment by AnasNeji** [comment]
@Quadrophonia I'm still curious as to how stopwords still manage to appear in topics even if they are ignored in the CountVectorizer level

For me adding the CountVectorizer with the stopwords param as shown in the code snippet above managed to remove them completely from the topics (Else, I had a completely seperate topic for stopwords)

@MaartenGr curious to hear what you think about this.

PS: You can also chain your topic representations to an openai generator (or any other LLM). This will generate a human-readable label based of the topic keywords and would likely help remove the stopwords

If that doesn't appeal to you you can basically hardcode remove the stopwords the code should look something like this :

```python
import nltk
from nltk.corpus import stopwords

# Download NLTK stopwords if not already present
nltk.download('stopwords', quiet=True)
stop_words = list(set(stopwords.words('english')))

for topic_id in topic_ids_list:
    if topic_id == -1:  # Skip outlier topic
        continue

    # Get top words for this topic - handle different return formats from BERTopic
    try:
        topic_data = topic_model.get_topic(topic_id)
        if topic_data:
            # Extract words from tuples (word, score) format
            topic_words = []
            for item in topic_data: 
                if isinstance(item, tuple) and len(item) >= 2:
                    topic_words.append(str(item[0]))  # word is first element
                elif isinstance(item, str):
                    topic_words.append(str(item))
                else:
                    logger.warning(f"Unexpected topic data format for topic {topic_id}: {item}")

            if topic_words:  # Only add non-empty topic word lists
                topic_words = [word for word in topic_words if word not in stop_words]
                topics_dict[topic_id] = topic_words
    except Exception as e:
        logger.error(f"Error processing topic {topic_id}: {e}")
        continue
```

**Comment by Quadrophonia** [comment]
@AnasNeji thank you. The reproducible example does not include the CountVectorizer, so while it might be helpful in removing the stopwords more generally, it is not the main focus of my post. My main concern is _not_ the stopwords. I am concerned about verifying the effect of chaining MMR (and, by extension but to a lesser extent, other representation models).

**Comment by MaartenGr** [comment]
MMR does a reranking of an initial set of keywords down to a smaller set of keywords. Therefore, any step that you do before that should have a larger `top_n_words` than its value that you set for MMR.
